### PR TITLE
IBX-2829: Added option to configure fields for register form

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | <https://github.com/ibexa/user/pull/40>
| Versions      | master, 4.2

Described added option to configure fields for register form

### CODE REVIEW

File: docs/guide/content_rendering/layout/add_register_user_template.md [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+You can create a registration form for users to your website by creating a twig template or editing the existing registration form in YAML file.
 Follow the instructions below to create and customize templates for a registration form, and a registration confirmation page.
 
 First, make sure you [enabled user registration](../../permission_use_cases.md#register-users).
 
-Next, in the `config/packages/views.yaml` file add the following configuration:
+## Configure existing form
+
+In the `config/packages/views.yaml` file specify the fields that should be part of your registration form under `allowed_field_definitions_identifiers`.
+You must also define if you are creating a backend user `user` or a frontend user `customer`under `user_type_identifier`.
```
[the remark this answers is not in the document]
Comment (MagdalenaZuba) [reply docs/guide/content_rendering/layout/add_register_user_template.md]: @ViniTou corrected

File: docs/guide/content_rendering/layout/add_register_user_template.md [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
+You can create a registration form for users to your website by creating a Twig template or editing the existing registration form in YAML file.
 Follow the instructions below to create and customize templates for a registration form, and a registration confirmation page.
 
 First, make sure you [enabled user registration](../../permission_use_cases.md#register-users).
 
-Next, in the `config/packages/views.yaml` file add the following configuration:
+## Configure existing form
+
+In the `config/packages/views.yaml` file specify the fields that should be part of your registration form under `allowed_field_definitions_identifiers`.
+You must also define what kind of user you want to create under `user_type_identifier` e.g. frontend user.
```
Comment (MagdalenaZuba) [inline docs/guide/content_rendering/layout/add_register_user_template.md:11]:
```suggestion
You can also define what kind of user you want to create under `user_type_identifier` e.g. frontend user.
```

File: docs/guide/content_rendering/layout/add_register_user_template.md [hunk 3]
```diff
@@ -1,11 +1,31 @@
...
-You can create a registration form for users to your website.
+You can create a registration form for users to your website by creating a Twig template or editing the existing registration form in YAML file.
 Follow the instructions below to create and customize templates for a registration form, and a registration confirmation page.
 
 First, make sure you [enabled user registration](../../permission_use_cases.md#register-users).
 
-Next, in the `config/packages/views.yaml` file add the following configuration:
+## Configure existing form
+
+In the `config/packages/views.yaml` file specify the fields that should be part of your registration form under `allowed_field_definitions_identifiers`.
```
Comment (DominikaK) [inline docs/guide/content_rendering/layout/add_register_user_template.md:10]:
```suggestion
In the `config/packages/views.yaml` file, under `allowed_field_definitions_identifiers`, specify the fields that should be part of your registration form.
```

File: docs/guide/content_rendering/layout/add_register_user_template.md [hunk 4]
```diff
@@ -1,11 +1,31 @@
...
+You can also define what kind of user you want to create under `user_type_identifier` e.g. frontend user.
```
Comment (DominikaK) [inline docs/guide/content_rendering/layout/add_register_user_template.md:11]: Link to the user types listed in the next page.

File: docs/guide/user_management/user_management.md [hunk 5]
```diff
@@ -134,6 +134,20 @@ It will check all User accounts in the database and list those that do not fit t
 
 You can allow your users to create accounts by employing the `/register` route. This route leads to a registration form that, when filled in, creates a new User Content item in the repository.
 
+### User types
+
+To decide to what workspace the user should be registered to, you need to specify their user type under `user_type_identifier`.
+Currently, there are two user types defined: `users` and `customers`.
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:140]:
```suggestion
There are two user types defined: `users` and `customers`.
```

File: docs/guide/user_management/user_management.md [hunk 6]
```diff
@@ -134,6 +134,20 @@ It will check all User accounts in the database and list those that do not fit t
 
 You can allow your users to create accounts by employing the `/register` route. This route leads to a registration form that, when filled in, creates a new User Content item in the repository.
 
+### User types
+
+To decide to what workspace the user should be registered to, you need to specify their user type under `user_type_identifier`.
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:139]: We don't use "workspace" anywhere else, either the term needs explanation, or rephrase.

File: docs/guide/user_management/user_management.md [hunk 7]
```diff
@@ -134,6 +134,20 @@ It will check all User accounts in the database and list those that do not fit t
...
+Currently, there are two user types defined: `users` and `customers`.
+`users` are backend users that are involved in creating the page such as editors and `customers` are frontend users.
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:141]:
```suggestion
`users` are Back Office users that are involved in creating the page such as editors, and `customers` are frontend users.
```

File: docs/guide/user_management/user_management.md [hunk 8]
```diff
@@ -134,6 +134,20 @@ It will check all User accounts in the database and list those that do not fit t
...
+
+```yaml
+ibexa:
+    system:
+        <siteaccess>:
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:146]:
```suggestion
        <scope>:
```

File: docs/guide/user_management/user_management.md [hunk 9]
```diff
@@ -146,9 +160,26 @@ ibexa:
...
 ```
 
+### Registration form field configuration
+
+To modify the registration form template add or remove fields under `allowed_field_definitions_identifiers`:
+
+```yaml
+ibexa:
+    system:
+        <siteaccess>:
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:170]:
```suggestion
        <scope>:
```

File: docs/guide/user_management/user_management.md [hunk 10]
```diff
@@ -146,9 +160,26 @@ ibexa:
                 group_id: <userGroupContentId>
 ```
 
+### Registration form field configuration
+
+To modify the registration form template add or remove fields under `allowed_field_definitions_identifiers`:
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:165]:
```suggestion
To modify the registration form template, add or remove fields under `allowed_field_definitions_identifiers`:
```

File: docs/guide/user_management/user_management.md [hunk 11]
```diff
@@ -146,9 +160,26 @@ ibexa:
...
+                    allowed_field_definitions_identifiers:
+                        - first_name
+                        - last_name
+                        - user_account
+```
+
 ### Other user management templates
 
-You can also modify the following form templates:
+You can also modify form templates in this way:
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:182]:
```suggestion
You can also modify form templates in the following way:
```

**Review by DominikaK — CHANGES_REQUESTED** [review]

**Pull request merged**